src/main/java/com/wiley/autotest/selenium/elements/upgrade/FindElementsLocator.java

```diff
@@ -28,11 +26,10 @@ public FindElementsLocator(WebDriver driver, By by, int index) {
         this.index = index;
     }
 
-
     @Override
     public WebElement find() {
         try {
-            return driver != null ? driver.findElements(by).get(index) : searchContext.findElements(by).get(index);
+            return driver != null ? driver.findElements(by).get(index) : searchContext.domElements(by).get(index).getWrappedWebElement();
         } catch (IndexOutOfBoundsException e) {
             throw new NoSuchElementException("Unable to find element " + by + ", Exception - " + e);
```

Comment:
What if add 'index' to the error message?

Reply:
Good idea, will add.